services/s3/src/main/java/software/amazon/awssdk/services/s3/handlers/EnableChunkedEncodingInterceptor.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3.handlers;
+
+import software.amazon.awssdk.annotations.ReviewBeforeRelease;
+import software.amazon.awssdk.core.SdkRequest;
+import software.amazon.awssdk.core.interceptor.Context;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.interceptor.ExecutionInterceptor;
+import software.amazon.awssdk.services.s3.auth.S3ExecutionAttributes;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.UploadPartRequest;
+
+/**
+ * Interceptor to enable chunked encoding on specific upload operations.
+ */
+@ReviewBeforeRelease("Check if there are other operations that should have chuncked encoding enabled by default")
+public class EnableChunkedEncodingInterceptor implements ExecutionInterceptor {
+
+    @Override
+    public SdkRequest modifyRequest(Context.ModifyRequest context, ExecutionAttributes executionAttributes) {
+        SdkRequest sdkRequest = context.request();
+
+        if (sdkRequest instanceof PutObjectRequest || sdkRequest instanceof UploadPartRequest) {
+            executionAttributes.putAttribute(S3ExecutionAttributes.ENABLE_CHUNKED_ENCODING, Boolean.TRUE);
```

Comment:
How would a customer explicitly disable this?

Reply:
We currently don't expose an option to set these values. So this interceptor might not even be needed as the default value is always true. I have create a issue for pre-GA launch to identify the correct behavior S3 wants.

Reply:
We got  a request in V1 to disable this since some alternative S3 implementations don't support it.